### PR TITLE
feat: Add ContainerSpec and ResourceSpec to Step proto def

### PR DESCRIPTION
**Pull request by terrytangyuan**
Signed-off-by: terrytangyuan <terrytangyuan@gmail.com>

### What changes were proposed in this pull request?

Add ContainerSpec and ResourceSpec to Step proto def.

### Why are the changes needed?

K8s resource needs to be supported as well.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Not tested.

### CODE REVIEW

**Comment by terrytangyuan** [comment]
cc @typhoonzero @Yancey1989

File: proto/couler.proto [hunk 1]
```diff
@@ -27,12 +27,23 @@ message StepIO {
   }
 }
 
+message ContainerSpec {
+    string image = 1;
+    repeated string command = 2;
```
Comment (ywskycn) [inline proto/couler.proto:32]: env?

Comment (terrytangyuan) [reply proto/couler.proto:32]: Thanks. Added.

Comment (ywskycn) [reply proto/couler.proto:32]: it is a key-value mapping?

Comment (terrytangyuan) [reply proto/couler.proto:32]: Fixed.

**Review by ywskycn — APPROVED** [review]

**Pull request merged**